libmamba/src/api/install.cpp

```diff
@@ -519,6 +519,13 @@ namespace mamba
             if (ctx.freeze_installed)
                 Console::print("Possible hints:\n  - 'freeze_installed' is turned on\n");
 
+            if (ctx.json)
+            {
+                Console::instance().json_write(
+                    { { "success", false }, { "solver_problems", solver.problems_to_json() } });
+                Console::instance().json_print();
```

Comment:
This is required to actually print the above json. Should this be somewhere "higher" near the main function? Maybe just always `Console::instance().json_print()` at the end of the main function?

Reply:
The current implementation clearly makes it hard to ensure that we don't print multiple times JSON outputs.
In the same time, this `libmamba` code can be used from other interfaces than `micromamba` so having a clean API to get/print the JSON from the buffer would be pretty nice. Then rely on the client interfaces (cli etc) to effectively display, write to a file, ...

notes: 
- we need to check all the places where `exit` is invoked from `libmamba` and remove them. it's up to the caller to handle that
- we catch errors thrown to nicely exit with a `critical` log message, we could also use that hook to return a JSON message there

Reply:
So should I leave it like this for now?

Reply:
Yeah, I think so. 
In the future, we might want to emulate what vcpkg-tool does. They have some "special" error handling functions, such as `exit_with_message` that we could use instead of plain `exit`, and that could take care of teardown functions such as logging json to stdout.

https://github.com/microsoft/vcpkg-tool/blob/8553f677a56aff07b4d5485fc3f1539546f28964/include/vcpkg/base/checks.h#L12-L19